build-tools/webpack/assets-writer-plugin.js

```diff
@@ -82,10 +82,8 @@ Object.assign( AssetsWriter.prototype, {
 
 			statsToOutput.chunks = stats.chunks.map( ( chunk ) =>
 				Object.assign( {}, chunk, {
+					chunks: stats.namedChunkGroups[ chunk.id ]?.chunks,
```

Comment:
This line causes the problem with missing chunks in production. The `chunk.id` is a chunk ID, but `stats.namedChunkGroups` uses chunk names as keys. In development they are both identical (we use `chunkIds: 'named'`) but in production the chunk IDs are numbers.

Reply:
I fixed this by a radical refactoring of the `assets.json` file format 🙂 It turns out that everything the app needs to serve the right assets is in the `stats.namedChunkGroups` object and in the `assets` field of this object's entries. For each chunk group, `assets` is a combined array of all assets from all chunks that comprise a chunk group. We don't need to look at `chunks` of the chunk group and then find all individual chunks one by one and read their `assets`. It's all already prepared by webpack at one place.

In the server code that reads the `assets.json` file, I also unified the two functions `getFilesForEntrypoint` and `getFilesForChunk`. Because entrypoint is just a special type of a chunk group. We only need one function, `getFilesForChunkGroup`.